tests/test_workers/test_pkg_managers/test_yarn.py

```diff
@@ -28,6 +28,7 @@
     "version": "1.0.0",
     "dependencies": {"chai": "^4.2.0", "fecha": HTTP_DEP_URL},
     "devDependencies": {"leftpad": GIT_DEP_URL},
+    "optionalDependencies": {"get-func-name": "^2.0.0"},
```

Comment:
Minor thing: A dependency specified by a normal version would never be replaced in package.json. Only external dependencies get replaced - https deps, git deps etc. Doesn't really matter for unit test data, but it might be less confusing if you use a URL instead of a version.